pkg/provider/provider.go

```diff
@@ -166,6 +170,80 @@ func (k *kubeProvider) invalidateResources() {
 
 // CheckConfig validates the configuration for this provider.
 func (k *kubeProvider) CheckConfig(ctx context.Context, req *pulumirpc.CheckRequest) (*pulumirpc.CheckResponse, error) {
+	urn := resource.URN(req.GetUrn())
+	label := fmt.Sprintf("%s.CheckConfig(%s)", k.label(), urn)
+	glog.V(9).Infof("%s executing", label)
+
+	olds, err := plugin.UnmarshalProperties(req.GetOlds(), plugin.MarshalOptions{
+		Label:        fmt.Sprintf("%s.olds", label),
+		KeepUnknowns: true,
+		SkipNulls:    true,
+		RejectAssets: true,
+	})
+	if err != nil {
+		return nil, pkgerrors.Wrapf(err, "CheckConfig failed because of malformed resource inputs")
+	}
+	news, err := plugin.UnmarshalProperties(req.GetNews(), plugin.MarshalOptions{
+		Label:        fmt.Sprintf("%s.news", label),
+		KeepUnknowns: true,
+		SkipNulls:    true,
+		RejectAssets: true,
+	})
+	if err != nil {
+		return nil, pkgerrors.Wrapf(err, "CheckConfig failed because of malformed resource inputs")
+	}
+
+	truthyValue := func(argName resource.PropertyKey, props resource.PropertyMap) bool {
+		if arg := props[argName]; arg.HasValue() {
+			switch {
+			case arg.IsString() && len(arg.StringValue()) > 0:
+				return true
+			case arg.IsBool() && arg.BoolValue():
+				return true
+			default:
+				return false
+			}
+		}
+		return false
+	}
+
+	renderYamlEnabled := truthyValue("renderYamlToDirectory", olds) ||
+		truthyValue("renderYamlToDirectory", news)
```

Comment:
Why do we turn this on even when it's only set in the `olds`?

Reply:
Good catch; that's incorrect. This bug wasn't ever hit in practice because `CheckConfig` would force replacement of the resource when the `renderYamlToDirectory` option changed.